observer.py

```diff
@@ -75,64 +87,67 @@ async def run_alerts():
-                                publisher = price_comp.publisher_key.key
-
-                                price.quoters[publisher] = price_comp.latest_price_info
-                                price.quoter_aggregates[
-                                    publisher
-                                ] = price_comp.last_aggregate_price_info
-
-                                if args.enable_prometheus:
-                                    gprice.labels(
-                                        symbol=symbol,
-                                        publisher=publisher,
-                                        status=price.quoters[publisher].price_status.name,
-                                    ).set(price.quoters[publisher].price)
-
-                            # Where the magic happens!
-                            price_errors = validators[symbol].verify_price(
-                                price=price, include_noisy=args.include_noisy_alerts
-                            )
-                            if price_errors:
-                                errors.extend(price_errors)
-
-                        # Send all notifications for a given symbol pair
-                        await validators[symbol].notify(
-                            errors,
-                            slack_webhook_url=args.slack_webhook_url,
-                            notification_mins=args.notification_snooze_mins,
+                    coingecko_price = coingecko_prices.get(product.attrs['base'])
+
+                    # prevent adding duplicate symbols
+                    if symbol not in validators:
+                        # TODO: If publisher_key is not None, then only do validation for that publisher
+                        validators[symbol] = PriceValidator(
+                            key=args.publisher_key,
+                            network=args.network,
+                            symbol=symbol,
+                            coingecko_price=coingecko_price
+                        )
+                    prices = await product.get_prices()
+
+                    for _, price_account in prices.items():
+                        price = Price(
+                            slot=price_account.slot,
+                            aggregate=price_account.aggregate_price_info,
+                            product_attrs=product.attrs,
+                            publishers=publishers,
+                        )
+                        price_account_errors = validators[symbol].verify_price_account(
+                            price_account=price_account,
+                            coingecko_price=coingecko_price,
+                        )
+                        if price_account_errors:
+                            filtered_errors = filter_errors(
+                                args.ignore, price_account_errors) if args.ignore else price_account_errors
+                            errors.extend(filtered_errors)
```

Comment:
nitpick: this filtering logic could be written in a single location if you saved `errors` here and then filtered around line 145 before calling notify.

Reply:
ah you're right, that's way cleaner